pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,182 @@
+		return errors.New("please supply a valid gitops Quick Start URL or name in --quickstart-profile")
+	}
+
+	if opts.gitOptions.URL == "" {
+		return errors.New("please supply a valid --git-url argument")
+	}
+	if opts.gitPrivateSSHKeyPath != "" && !file.Exists(opts.gitPrivateSSHKeyPath) {
+		return errors.New("please supply a valid --git-private-ssh-key-path argument")
+	}
+
+	quickstartRepoURL, err := repoURLForQuickstart(opts.quickstartNameArg)
+	if err != nil {
+		return errors.Wrapf(err, "please supply a valid Quick Start name or URL")
+	}
+
+	if err := cmdutils.NewGitopsMetadataLoader(cmd).Load(); err != nil {
+		return err
+	}
+	cfg := cmd.ClusterConfig
+	ctl, err := cmd.NewCtl()
+	if err != nil {
+		return err
+	}
+
+	if err := ctl.CheckAuth(); err != nil {
+		return err
+	}
+	if err := ctl.RefreshClusterConfig(cfg); err != nil {
+		return err
+	}
+	kubernetesClientConfigs, err := ctl.NewClient(cfg)
+	if err != nil {
+		return err
+	}
+	k8sConfig := kubernetesClientConfigs.Config
+
+	k8sRestConfig, err := clientcmd.NewDefaultClientConfig(*k8sConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return errors.Wrap(err, "cannot create Kubernetes client configuration")
+	}
+	k8sClientSet, err := kubeclient.NewForConfig(k8sRestConfig)
+	if err != nil {
+		return errors.Errorf("cannot create Kubernetes client set: %s", err)
+	}
+
+	// Create the flux installer. It will clone the user's repository in the outputPath
+	fluxOpts := flux.InstallOpts{
+		GitURL:      opts.gitOptions.URL,
+		GitBranch:   opts.gitOptions.Branch,
+		GitEmail:    opts.gitOptions.Email,
+		GitUser:     opts.gitOptions.User,
+		Namespace:   "flux",
+		GitFluxPath: "flux/",
+		WithHelm:    true,
+		Timeout:     git.DefaultGitTimeout,
+	}
+	fluxInstaller := flux.NewInstaller(context.Background(), k8sRestConfig, k8sClientSet, &fluxOpts)
+
+	processor := &fileprocessor.GoTemplateProcessor{
+		Params: fileprocessor.NewTemplateParameters(cmd.ClusterConfig),
+	}
+
+	// Create the profile generator. It will output the processed templates into a new "/base" directory into the user's repo
+	usersRepoName, err := git.RepoName(opts.gitOptions.URL)
+	if err != nil {
+		return err
+	}
+	usersRepoDir := filepath.Join(opts.outputPath, usersRepoName)
+	profileOutputPath := filepath.Join(usersRepoDir, "base")
+
+	profile := &gitops.Profile{
+		Processor: processor,
+		Path:      profileOutputPath,
+		GitOpts: git.Options{
+			URL:    quickstartRepoURL,
+			Branch: "master",
+		},
+		GitCloner: git.NewGitClient(context.Background(), git.ClientParams{
+			Timeout: git.DefaultGitTimeout,
+		}),
+		FS: afero.NewOsFs(),
+		IO: afero.Afero{Fs: afero.NewOsFs()},
+	}
+
+	// A git client that operates in the user's repo
+	gitClient := git.NewGitClient(context.Background(), git.ClientParams{
+		PrivateSSHKeyPath: opts.gitPrivateSSHKeyPath,
+		Timeout:           git.DefaultGitTimeout,
+		Dir:               usersRepoDir,
+	})
+
+	gitOps := gitops.GitOps{
```

Comment:
Suggestions for new names for the struct are welcome. GitopsApplier?

Reply:
Given the package is `gitops` (and the file `apply.go`) then maybe just `Applier`?
(That way, it would read as `gitops.Applier` when called elsewhere.)